basxconnect/core/views/person/person_modals_views.py

```diff
@@ -156,13 +157,32 @@ def get_layout(self):
 class EditPostalAddressView(EditView):
     fields = ["type", "address", "postcode", "city", "country"]
 
-    def get_success_url(self):
-        return reverse_model(
-            self.object.person, "read", kwargs={"pk": self.object.person.pk}
+    def get_form(self, form_class=None):
+        ret = super().get_form(form_class)
+        ret.fields["is_primary"] = django.forms.BooleanField(
+            label=_("Use as primary postal address"), required=False
         )
+        return ret
+
+    def post(self, request, *args, **kwargs):
+        ret = super().post(request, *args, **kwargs)
+        is_primary = request.POST.get("is_primary")
```

Comment:
In Django we should never use the POST or GET dictionary directly, except we have a good reason (like not using a Django form and verifying the data). The problem is that HTTP-payload is unverfied user-data and therefore needs to be parsed and verified. That is the main reason that we use Django forms and, in this instance, ```django.forms.BooleanField``` at all.
Form data should always be accessed via the form.cleaned_data dictionary. This can be used for GET POST data. It will also take care of handling that ```"on"``` string and convert it properly to a boolean value.